www/nginx/src/opnsense/scripts/nginx/ngx_auth.php

```diff
@@ -45,15 +45,21 @@ function password_auth_test($username, $password, $auth_server)
 
 function password_auth($auth_server = 'Local Database')
 {
-    if (!isset($_SERVER['PHP_AUTH_PW']) || !isset($_SERVER['PHP_AUTH_PW'])) {
+    if (!isset($_SERVER['PHP_AUTH_PW']) || !isset($_SERVER['PHP_AUTH_USER'])) {
         return false;
     }
     return password_auth_test($_SERVER['PHP_AUTH_USER'], $_SERVER['PHP_AUTH_PW'], $auth_server);
 }
 
-if (password_auth()) {
+if (empty($_SERVER['AUTH_SERVER'])) {
+    $auth_server = 'Local Database';
+} else {
+    $auth_server = $_SERVER['AUTH_SERVER'];
+}
+
+if (password_auth($auth_server)) {
     header("HTTP/1.1 200 OK");
 } else {
     header("HTTP/1.1 401 Authorization Required");
```

Comment:
Isn't there a 4xx for bad credentials?

Reply:
401 means I need a username / password to proceed. In a browser it will show to a authentication popup. There is no separate call which could be used similar except 403 forbidden or 402 payment required (does not fit well here anyway) in HTTP. Please also keep in mind that this is an internal API call, which needs to follow the specification: https://nginx.org/en/docs/http/ngx_http_auth_request_module.html